File: Auth/QiniuToken/QiniuToken.php
<?php
require_once __DIR__ . '/../autoload.php';
use Qiniu\Auth;
use Qiniu\Http\Client;
  
$accessKey = 'ak';
$secretKey = 'sk';
  
$auth = new Auth($accessKey, $secretKey);
$data=array();
$url="xxxx";
$token=$auth->authorizationV2($url, POST,json_encode($data), 'application/json')['Authorization'];
echo $token."\n";
  
$client = new Client();
$headers=array(
    'Authorization' => $token,
    'Content-Type' => 'application/json'
);
  
$res=$client->post($url, json_encode($data), $headers);
echo $res->statusCode."\n";
echo print_r($res->headers)."\n";
echo $res->body."\n";
?>

File: Auth/QiniuToken/QiniuToken.py
# -*- coding: utf-8 -*-
from qiniu import Auth, QiniuMacAuth
import requests, json, sys
 
access_key = '<ak>'
secret_key = '<sk>'
  
auth = QiniuMacAuth(access_key, secret_key)
  
url = '<url>'
body = {
}
 
token = "Qiniu " + auth.token_of_request('POST', '<host>', url, "", body=json.dumps(body), content_type='application/json')
print token
 
r = requests.post(url, data=json.dumps(body), headers={'Authorization': token, 'Content-Type': 'application/json'})
 
print r
print r.text
